wgsl/index.bs

```diff
@@ -501,18 +520,49 @@ See [[#keyword-summary]] for the list of WGSL keywords.
 An <dfn>identifier</dfn> is a kind of [=token=] used as a name.
 See [[#declaration-and-scope]] and [[#directives]].
 
-The form of an identifier is defined via pattern-matching, except that
-an identifier:
-* must not have the same spelling as a [=keyword=] or as a [=reserved word=].
-* must not be `_` (a single underscore)
-* must not start with two underscores.
+The form of an identifier is based on the
+[=Unicode Standard Annex #31 for Unicode Version 14.0.0|Unicode Standard Annex #31=] for
+[[!UnicodeVersion14|Unicode Version 14.0.0]],
+with the following elaborations. Identifiers do not compare under canonical equivalence
+and they are not normalized otherwise,
+meaning that single accented code points are distinct from the same characters constructed
+by an accent and a letter, and Unicode code points that look similar are not
+treated as the same and they compare by their code points.
+
+Identifiers use the following profile described in terms of [=UAX31 Grammar=]:
+
+```
+<Identifier> := <Start> <Continue>* (<Medial> <Continue>+)*
+
+<Start> := XID_Start + U+005F
+<Continue> := <Start> + XID_Continue
+<Medial> :=
+```
+
+This means identifiers with non-ASCII characters like these are
+valid: `Δέλτα`, `réflexion`, `Кызыл`, `𐰓𐰏𐰇`, `朝焼け`, `سلام`, `검정`, `שָׁלוֹם`, `गुलाबी`, `փիրուզ`.
+
+With the following exceptions:
+* An identifier must not have the same spelling as a [=keyword=] or as a [=reserved word=].
+* An identifier must not be `_` (a single underscore)
+* An identifier must not start with two underscores.
 
 <div class='syntax' noexport='true'>
   <dfn for=syntax>ident</dfn> :
 
-    | `/([a-zA-Z_][0-9a-zA-Z][0-9a-zA-Z_]*)|([a-zA-Z][0-9a-zA-Z_]*)/`
+    | `/([\p{XID_Start}])|([_\p{XID_Start}][\p{XID_Continue}]+)/`
```

Comment:
This does not allow the identifer `x_`

I think this needs an underscore to be allowed beside the XID_Continue.


            ([\p{XID_Start}])|([_\p{XID_Start}][_\p{XID_Continue}]+)

If you are trying to exclude the _ and __ cases then I think you would need to break out more cases.

             \p{XID_Start}
             [_\p{XID_Start}][\p{XID_Continue}][_\p{XID_Continue}]*
             [\p{XID_Start}][_\p{XID_Continue}]+

Reply:
XID_Continue includes _ (low line), that's why it is not additionally included. So that kind of makes it impossible to do exception for double underscore case... :\

Reply:
@dneto0 I applied this latest regex, hopefully it does capture cases like x_ in your test RegEx engine too.

Reply:
(just for anybody reading and having issues, the uy suffix is important in JS RegEx and in Python, you should probably use the Unicode RegEx package and not the built-in)

Reply:
I guess that can be simplified to avoid overlap between the second and third cases:

         \p{XID_Start}
         [_\p{XID_Start}][\p{XID_Continue}][_\p{XID_Continue}]*
         [\p{XID_Start}]_[_\p{XID_Continue}]*

Reply:
@dneto0 The second one with omission of low-line in second character I think tries to block starting with double underscore, but it will still pass because even if there is no underscore there, XID_Continue does have support for underscore intrinsically, and unfortunately support for Unicode Set operations are not universal in RegEx engines. And XID_Start, which does actually omit low-line underscore, can not substitute XID_Continue for a second character.

Reply:
Unicode provided listing of XID_Continue content: https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5B%3AXID_Continue%3A%5D&g=&i=

Reply:
What are the uy modifiers?  I can't find documentation for them.

Reply:
'y' modifier is: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/sticky 
'u' modifier is for unicode:  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/unicode

Reply:
And yeah, I didn't realize the XID_Continue set included underscore "Low line" until I looked.
So my suggestions were incorrect.